src/pip/_internal/pyproject.py

```diff
@@ -58,7 +58,7 @@ def load_pyproject_toml(
     has_setup = os.path.isfile(setup_py)
 
     if has_pyproject:
-        with open(pyproject_toml, encoding="utf-8") as f:
+        with open(pyproject_toml, "rb") as f:
```

Comment:
Please avoid merging this without an OK from me -- I haven't had time to take a look at what the issue is on the TOML repository, but this is definitely not the direction to take IMO.